### DISCUSSION TITLE
Error during preparing data

### DISCUSSION THREAD

**Discussion by SnowHana** [dcaribou/transfermarkt-datasets#313] [Q&A] [answered]
Hi hope you well! 
So after installing and running `poetry shell` and running `make dvc_pull`, I tried to prepare data to get a csv files by running `make prepare_local` command. Test runs smoothly except one test called 
>*test not_null_transfers_player_name (models/curated/models.yml)*. 

I tried in both github codespace and local, and both are producing the exact same error.
Below is the log.

Thank you in advance!

(Also, is  `make prepare_local` the command that produces the csv files that's uploaded in kaggle? If not, is there a way to produce a csv file identical to that in kaggle?)

---

$ make prepare_local target
cd dbt && dbt deps && dbt build --threads 4 --target dev
07:40:05  Running with dbt=1.7.3
07:40:06  Installing calogica/dbt_expectations
07:40:07  Installed from version 0.9.0
07:40:07  Updated version available: 0.10.4
07:40:07  Installing dbt-labs/dbt_utils
07:40:07  Installed from version 1.1.1
07:40:07  Updated version available: 1.3.0
07:40:07  Installing calogica/dbt_date
07:40:07  Installed from version 0.8.1
07:40:07  Updated version available: 0.10.1
07:40:07  
07:40:07  Updates available for packages: ['calogica/dbt_expectations', 'dbt-labs/dbt_utils', 'calogica/dbt_date']                 
Update your versions in packages.yml, then run dbt deps
07:40:10  Running with dbt=1.7.3
07:40:11  Registered adapter: duckdb=1.7.0
07:40:11  Found 18 models, 66 tests, 8 sources, 0 exposures, 0 metrics, 746 macros, 0 groups, 0 semantic models
07:40:11  
07:40:11  Concurrency: 4 threads (target='dev')
07:40:11  
07:40:11  1 of 84 START sql table model dev.base_clubs ................................... [RUN]
07:40:11  2 of 84 START sql table model dev.base_competitions ............................ [RUN]
07:40:11  3 of 84 START sql table model dev.base_game_events ............................. [RUN]
07:40:11  4 of 84 START sql table model dev.base_game_lineups ............................ [RUN]
07:40:11  2 of 84 OK created sql table model dev.base_competitions ....................... [OK in 0.30s]
....
7:40:58  67 of 84 START test not_null_transfers_player_name ............................. [RUN]
07:40:58  66 of 84 PASS not_null_transfers_player_id ..................................... [PASS in 0.05s]
07:40:58  68 of 84 START test not_null_transfers_to_club_id .............................. [RUN]
07:40:58  67 of 84 FAIL 254 not_null_transfers_player_name ............................... [FAIL 254 in 0.06s]
...
07:42:15  83 of 84 PASS not_null_appearances_game_id ..................................... [PASS in 0.08s]
07:42:16  84 of 84 PASS unique_appearances_appearance_id ................................. [PASS in 0.25s]
07:42:16  
07:42:16  Finished running 18 table models, 66 tests in 0 hours 2 minutes and 4.81 seconds (124.81s).
07:42:16  
07:42:16  Completed with 1 error and 0 warnings:
07:42:16  
07:42:16  Failure in test not_null_transfers_player_name (models/curated/models.yml)
07:42:16    Got 254 results, configured to fail if != 0
07:42:16  
07:42:16    compiled Code at target/compiled/transfermarkt_datasets/models/curated/models.yml/not_null_transfers_player_name.sql
07:42:16  
07:42:16  Done. PASS=83 WARN=0 ERROR=1 SKIP=0 TOTAL=84
make: *** [Makefile:67: prepare_local] Error 1

**Comment by dcaribou** [comment]
Yeah,  this error is showing up from time to time lately. Not sure why. I've created this issue for looking into it at some point, you might want to subscribe to it https://github.com/dcaribou/transfermarkt-datasets/issues/314

> Also, is make prepare_local the command that produces the csv files that's uploaded in kaggle? If not, is there a way to produce a csv file identical to that in kaggle?

Yes, prepare_local will produce the data that is pushed to Kaggle in `data/prep` folder

**Reply by dcaribou** [reply, marked as answer]
If you get the latest change from master now, you should no longer have the error.

**Reply by SnowHana** [reply]
Thanks it works smoothly now!
